libsignal-service/src/sender.rs

```diff
@@ -139,12 +139,14 @@ where
                             ),
                         )?;
                     }
-                    self.cipher.store_context.delete_session(
-                        &libsignal_protocol::Address::new(
-                            &recipient.e164,
-                            *extra_device_id,
-                        ),
-                    )?;
+                    if let Some(ref e164) = recipient.e164 {
```

Comment:
Sessions with Uuid should be deleted too, no? Maybe `get_preferred_protocol_address`?

Reply:
Yes, it was already done before? (open the code above).

Reply:
Yep, missed that, sorry for that.